org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/participants/XSDDocumentLinkParticipant.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.lemminx.extensions.xsd.participants;
+
+import static org.eclipse.lemminx.utils.XMLPositionUtility.createDocumentLink;
+
+import java.util.List;
+
+import com.google.common.base.Strings;
+
+import org.apache.xerces.impl.XMLEntityManager;
+import org.apache.xerces.util.URI.MalformedURIException;
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.dom.DOMAttr;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.dom.DOMRange;
+import org.eclipse.lemminx.services.extensions.IDocumentLinkParticipant;
+import org.eclipse.lsp4j.DocumentLink;
+
+public class XSDDocumentLinkParticipant implements IDocumentLinkParticipant {
+
+	@Override
+	public void findDocumentLinks(DOMDocument document, List<DocumentLink> links) {
+		DOMElement root = document.getDocumentElement();
+		if (root == null || !root.getLocalName().equals("schema")) {
+			return;
+		}
+		String xmlSchemaPrefix = root.getPrefix();
+		List<DOMNode> children = root.getChildren();
+		for (DOMNode child : children) {
+			if (child.isElement() && child.getLocalName().equals("include")
+					&& child.getPrefix().equals(xmlSchemaPrefix)) {
```

Comment:
To prevent from NPE with child.getPrefix().equals, I will suggest to write `Objects.equals(child.getPrefix(), xmlSchemaPrefix)` by using `java.util.Objects`